webclient/projekt/signals.py

```diff
@@ -11,6 +12,22 @@
 
 
 @receiver(pre_save, sender=Projekt)
+def projekt_pre_save(sender, instance, **kwargs):
+    create_projekt_vazby(sender, instance)
+    change_termin_odevzdani_NZ(sender, instance)
+
+def change_termin_odevzdani_NZ(sender, instance, **kwargs):
+    try:
```

Comment:
Navrhuji úpravu
```
    try:
        instance_db = sender.objects.get(pk = instance.pk)
    except sender.DoesNotExist:
        instance_db = None
        
    if instance.termin_odevzdani_nz == instance_db.termin_odevzdani_nz:
        if instance.datum_ukonceni != instance_db.datum_ukonceni:
            logger.debug("Zmene datum ukonceni a nezmenene termin_odevzdani_NZ - menim automaticky termin_odevzdani_NZ")
            instance.termin_odevzdani_nz = instance.datum_ukonceni
            instance.termin_odevzdani_nz.replace(year=instance.termin_odevzdani_nz.year + 3)
```
Přijde mi to takto jednodušší. Dej prosím vědět, zda souhlasíš.

Reply:
to by pak padalo pre create zaznamu, kvuli instance_db.termin_odevzdani_nz:

Reply:
Máš pravdu, přidal jsem ještě do podmínky instance_db na začátek, takže se nic dalšího kontrolovat nebude, pokud hledání instance spadne do výjimky a uloží se tam None.

Reply:
upraveno

Reply:
Super, díky moc!